rust/src/genesis/byron/config.rs

```diff
@@ -0,0 +1,126 @@
+/// use cardano_multiplatform_lib::genesis::byron::config::ProtocolMagic;
+///
+/// assert_eq!(ProtocolMagic::default(), ProtocolMagic::new(0x2D964A09));
+/// ```
+///
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Hash)]
+#[cfg_attr(feature = "generic-serialization", derive(Serialize, Deserialize))]
+#[repr(C)]
+pub struct ProtocolMagic(pub u32);
+impl ProtocolMagic {
+    #[deprecated]
+    pub fn new(val: u32) -> Self {
+        ProtocolMagic(val)
+    }
+}
+impl fmt::Display for ProtocolMagic {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+impl ::std::ops::Deref for ProtocolMagic {
+    type Target = u32;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+impl From<u32> for ProtocolMagic {
+    fn from(v: u32) -> Self {
+        ProtocolMagic(v)
+    }
+}
+impl Default for ProtocolMagic {
+    fn default() -> Self {
+        NetworkInfo::mainnet().protocol_magic()
+    }
+}
+impl cbor_event::se::Serialize for ProtocolMagic {
+    fn serialize<'se, W: Write>(
+        &self,
+        serializer: &'se mut Serializer<W>,
+    ) -> cbor_event::Result<&'se mut Serializer<W>> {
+        serializer.write_unsigned_integer(self.0 as u64)
+    }
+}
+impl cbor_event::Deserialize for ProtocolMagic {
+    fn deserialize<R: BufRead>(reader: &mut Deserializer<R>) -> cbor_event::Result<Self> {
+        let v = reader.unsigned_integer()? as u32;
+        Ok(ProtocolMagic::from(v))
+    }
+}
+
+/// Configuration for the wallet-crypto
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+#[cfg_attr(feature = "generic-serialization", derive(Serialize, Deserialize))]
+pub struct Config {
+    pub protocol_magic: ProtocolMagic,
+}
+impl Config {
+    pub fn new(protocol_magic: ProtocolMagic) -> Self {
+        Config {
+            protocol_magic: protocol_magic,
+        }
+    }
+}
+impl Default for Config {
+    fn default() -> Self {
+        Config::new(ProtocolMagic::default())
+    }
+}
+
+/// A subset of the genesis data. The genesis data is a JSON file
+/// whose canonicalized form has the hash 'genesis_prev', which is the
+/// parent of the genesis block of epoch 0. (Note that "genesis data"
+/// is something completely different from a epoch genesis block. The
+/// genesis data is not stored in the chain as a block.)
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct GenesisData {
+    // FIXME: genesis_prev shouldn't be here since it's computed *from* the GenesisData.
+    pub genesis_prev: BlockHeaderHash,
+    pub epoch_stability_depth: usize, // a.k.a. 'k'
+    pub start_time: SystemTime,
+    pub slot_duration: Duration,
+    pub protocol_magic: ProtocolMagic,
+    pub fee_policy: fees::LinearFee,
+    pub avvm_distr: BTreeMap<chain_crypto::PublicKey<Ed25519>, utils::Coin>, // AVVM = Ada Voucher Vending Machine
+    // TODO: convert to ByronAddress without affecting order
+    pub non_avvm_balances: BTreeMap<legacy_address::Addr, utils::Coin>,
```

Comment:
Note: this gives a different order than what db-sync gives. Not sure which one is correct

Reply:
https://github.com/input-output-hk/cardano-db-sync/issues/1107

Reply:
Does the order matter here? Does how it's stored here impact how the hash is computed (order of some map in CBOR?) It could be not important if some order is specified in that part of code and the hashes match up, even if the order here is not what db-sync gives. I don't know enough about the genesis block to really say what should happen though.